src/cargo/core/source/source_id.rs

```diff
@@ -293,7 +275,7 @@ impl SourceId {
 
     pub fn is_default_registry(&self) -> bool {
         match self.inner.kind {
-            Kind::Registry => {}
+            Kind::Registry if self.inner.registry_key.is_none() => {}
```

Comment:
Should this clause be removed? It seems like it may cause different behavior between:

```toml
a = "0.1"
b = { version = "0.1", registry = "crates-io" }